src/rgw/rgw_auth_s3.cc

```diff
@@ -1024,8 +1024,8 @@ get_v2_signature(CephContext* const cct,
 
   const auto digest = calc_hmac_sha1(secret_key, string_to_sign);
 
-  /* 64 is really enough */;
-  char buf[64];
+  /* Sized for signature */;
+  char buf[AWSEngine::VersionAbstractor::SIGNATURE_MAX_SIZE];
   const int ret = ceph_armor(std::begin(buf),
                              std::begin(buf) + 64,
```

Comment:
the `+ 64` should change too, right? i'm guessing that `std::end(buf)` would do the right thing without the need for pointer arithmetic

sorry for late review

Reply:
Yes, fixed it